wazimap_ng/points/admin/forms/category_admin_form.py

```diff
@@ -22,6 +27,23 @@ def __init__(self, *args, **kwargs):
                 self.fields["description"].initial = metadata.description
                 self.fields["licence"].initial = metadata.licence
 
+    def clean(self):
+        cleaned_data = super(CategoryAdminForm, self).clean()
+        document = cleaned_data.get('import_collection', None)
+        required_headers = ["name", "longitude", "latitude", "id", "status", "status last updated", "class", "volume",
+                            "discharges into"]
+        if document is not None:
+            headers = pd.read_csv(BytesIO(document.read()), nrows=1, dtype=str).columns.str.lower().str.strip()
+            missing_headers = [
+                h.capitalize() for h in list(set(required_headers) - set(headers))
```

Comment:
So this is where the magic happens. Removing headers provided from a list of required headers and then check if there are headers remaining from required_headers. Nice